src/Whitelist/VulnerabilityExcluder.spec.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import expect, { ossIndexServerResults } from '../Tests/TestHelper';
+import { filterVulnerabilities } from './VulnerabilityExcluder';
+import mock from 'mock-fs';
+import { createAppLogger } from '../Application/Logger/Logger';
+
+const jsonWithOnlyOneMatch = `{
+  "ignore": [ 
+    {
+      "id": "test_id"
+    }
+  ]
+}`;
+
+const json = `{
+  "ignore": [ 
+    {
+      "id": "test_id"
+    },
+    {
+      "id": "test_id2"
+    }
+  ]
+}`;
+
+const jsonWithNonApplicableId = `{
+  "ignore": [ {
+    "id": "qwepquepoqwe"
+    }
+  ]
+}`;
+
+const nonSensicalJson = `{
```

Comment:
This is non-sensical but I think it still is a valid json.
What about `const invalidJson = "asasa";`?

Reply:
> This is non-sensical but I think it still is a valid json.
> What about `const invalidJson = "asasa";`?

No, nevermind. I see now that it is indeed invalid. I'll see myself out.